src/lib.rs

```diff
@@ -174,16 +176,18 @@ impl Deref for LockedHeap {
 }
 
 #[cfg(feature = "use_spin")]
-unsafe impl<'a> Alloc for &'a LockedHeap {
-    unsafe fn alloc(&mut self, layout: Layout) -> Result<*mut u8, AllocErr> {
-        self.0.lock().allocate_first_fit(layout)
+unsafe impl<'a> GlobalAlloc for &'a LockedHeap {
+    unsafe fn alloc(&self, layout: Layout) -> *mut Opaque {
+        LOCKED_ALLOCATOR.0.lock().allocate_first_fit(layout).ok().map_or(0 as *mut Opaque, |allocation| {
```

Comment:
I think we should be able to do just `self.0.lock` without the static `LOCKED_ALLOCATOR`.

Reply:
Yep, Mutex lock() uses &self and returns MutexGuard<T>, so static is not needed here.

Reply:
Is the &'a needed in impl? It works without it.